src/components/Editor/DebugLine.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+import { Component } from "react";
+import type { SourceEditor } from "devtools-source-editor";
+import type { Frame } from "../../types";
+
+type DebugLineProps = {
+  editor: SourceEditor,
+  frame: Frame,
+  visibleSourceId: string
```

Comment:
Unfortunately, that distinction is made for a very particular reason. You can select a new sourceId but not have it yet be visible (if you're waiting for it the source to be fetched, for example). For this component to know when to run its editor commands, it needs to know when the source is actually displayed. Prior to this change, the highlight would run right away, and when the source arrived, it would be put in a new document and not get the highlight. This makes me think we need a way of telling side-effecting editor children something about the identity of the displayed document.